src/sdk/config_fs.js

```diff
@@ -265,104 +450,217 @@ class ConfigFS {
-     * @param {Object[]} [access_keys_to_delete]
+     * @param {string} owner_id
      * @returns {Promise<void>} 
      */
-    async delete_account_config_file(account_name, access_keys_to_delete = []) {
-        const account_config_path = this.get_account_path_by_name(account_name);
-        await native_fs_utils.delete_config_file(this.fs_context, this.accounts_dir_path, account_config_path);
-        for (const access_keys of access_keys_to_delete) {
-            const access_key_config_path = this.get_account_path_by_access_key(access_keys.access_key);
-            await nb_native().fs.unlink(this.fs_context, access_key_config_path);
+    async link_account_name_index(account_id, account_name, owner_id) {
+        const account_name_path = this.get_account_or_user_path_by_name(account_name, owner_id);
+        const account_id_relative_path = this.get_account_relative_path_by_id(account_id);
+        await nb_native().fs.symlink(this.fs_context, account_id_relative_path, account_name_path);
+    }
+
+    /**
+     * unlink_account_name_index unlinks the access key from the config directory
+     * 1. get the account name path
+     * 2. check realpath on the account name path
+     * 3. check if the account id path is the same as the account name path 
+     * 4. unlink the account name path
+     * 5. else, do nothing as the name path might already point to a new identity/deleted by concurrent calls 
+     * @param {string} account_name
+     * @returns {Promise<void>} 
+     */
+    async unlink_account_name_index(account_name, account_id_config_path) {
+        const account_name_path = this.get_account_path_by_name(account_name);
+        const full_path = await nb_native().fs.realpath(this.fs_context, account_name_path);
+        if (full_path === account_id_config_path) {
+            await nb_native().fs.unlink(this.fs_context, account_name_path);
+        }
+    }
+
+    //////////////////////////////////////
+    //////   ACCESS KEYS INDEXES    //////
+    //////////////////////////////////////
+
+    /**
+     * link_access_keys_index links the access keys symlinks
+     * @param {Object[]} access_keys_to_link
+     * @returns {Promise<void>} 
+     */
+    async link_access_keys_index(account_id, access_keys_to_link = []) {
+        if (!access_keys_to_link?.length) return;
+        const account_config_relative_path = this.get_account_relative_path_by_id(account_id);
+        for (const access_keys of access_keys_to_link) {
+            const new_access_key_path = this.get_account_or_user_path_by_access_key(access_keys.access_key);
+            await nb_native().fs.symlink(this.fs_context, account_config_relative_path, new_access_key_path);
         }
     }
 
     /**
-     * unlink_access_key_symlink unlinks the access key from the file system
+     * unlink_access_key_index unlinks the access key from the config directory
+     * 1. get the account access_key path
+     * 2. check realpath on the account name path
```

Comment:
I think it was copied
```suggestion
     * 2. check realpath on the access key path
```
Could you add in the comment why we checked that? (for example - to make sure that it belongs to an existing account)